app/services/finance/ecf/assurance_report/query.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module Finance
+  module ECF
+    class AssuranceReport
+      class Query
+        def initialize(statement)
+          self.statement = statement
+        end
+
+        def participant_declarations
+          ParticipantDeclaration::ECF.find_by_sql(sql)
+        end
+
+      private
+
+        attr_accessor :statement
+
+        def sql
+          <<~EOSQL
```

Comment:
could we add a test for this query service, as we're setting values through `case`?